testing/test_git.py

```diff
@@ -191,3 +191,31 @@ def test_git_feature_branch_increments_major(wd):
     wd("git checkout -b feature/fun")
     wd.commit_testfile()
     assert wd.get_version(version_scheme="python-simplified-semver").startswith("1.1.0")
+
+
+@pytest.mark.issue("https://github.com/pypa/setuptools_scm/issues/303")
+def test_exclude(wd):
+    wd.commit_testfile()
+    wd("git tag apache-arrow-0.11.1")
+    wd.commit_testfile()
+    wd("git tag apache-arrow-js-0.9.9")
+    wd.commit_testfile()
+    assert wd.get_version(
+        tag_regex=r"^apache-arrow-([\.0-9]+)$", exclude="*js*"
+    ).startswith(
+        "0.11.2"
+    )
+
+
+@pytest.mark.issue("https://github.com/pypa/setuptools_scm/issues/303")
+def test_match(wd):
+    wd.commit_testfile()
+    wd("git tag apache-arrow_0.11.1")
+    wd.commit_testfile()
+    wd("git tag apache-arrow-js_0.9.9")
+    wd.commit_testfile()
+    assert wd.get_version(
+        tag_regex=r"^apache-arrow_([\.0-9]+)$", match="apache-arrow_*"
```

Comment:
This feels redundant to me - I think the project should try hard not to require the user to specify both a tag_regex and a match just to avoid a bug in tag_regex.